test/behaviour/with_otp_app_key_test.exs

```diff
@@ -29,6 +56,22 @@ defmodule Behaviour.WithOtpAppKey do
     assert behaviour.__knigge__(:implementation) == SomeModule
   end
 
+  test "works fine with config_key path being set in the Application environment" do
+    Application.put_env(:knigge, :sys, module: SomeModule)
+
+    # Should not raise
+    behaviour =
+      defmodule_salted WorkingBehaviour do
+        use Knigge,
+          otp_app: :knigge,
+          config_key: [:sys, :module]
+      end
```

Comment:
Reading this test, I think it'd be beneficial for readability to deprecate `config_key` in favor of `config_path`. The `Options` module has a `map_deprecated` function ([link](https://github.com/sascha-wolf/knigge/blob/e0886d345c7ee9af5be716a33b78d4f358cc999b/lib/knigge/options.ex#L170-L175)) which we can use to map `config_key` to `config_path` while also printing a deprecation warning.

What do you think?

Reply:
Good idea. I will add this.